x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/details_page_mappings_content.tsx

```diff
@@ -475,18 +477,32 @@ export const DetailsPageMappingsContent: FunctionComponent<{
             {!index.hidden && (
               <EuiFlexItem grow={false}>
                 {!isAddingFields ? (
-                  <EuiButton
-                    onClick={addFieldButtonOnClick}
-                    iconType="plusInCircle"
-                    color="text"
-                    size="m"
-                    data-test-subj="indexDetailsMappingsAddField"
+                  <EuiToolTip
+                    position="bottom"
+                    content={
+                      hasUpdateMappingsPrivilege !== undefined && !hasUpdateMappingsPrivilege
```

Comment:
When it's a stack, `hasUpdateMappingsPrivilege` is undefined & we don't want privilege check too. So checking if `hasUpdateMappingsPrivilege` is not undefined explicitly. Added similar logic to delete document & edit settings

Reply:
seems like this is worth a comment in the code at very least